### PR TITLE
[JSDoc] Add missing documentation for botbuilder-dialogs-adaptive input folder

### PR DESCRIPTION
**Pull request by sw-joelmut**
Addresses # 2602

## Description
This PR adds all the missing JSDoc documentation for [botbuilder-dialogs-adaptive](https://github.com/microsoft/botbuilder-js/tree/main/libraries/botbuilder-dialogs-adaptive) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet).

## Specific Changes

- Ported documentation from botbuilder-dotnet
- Added `@private` for private methods.
- Added `@protected` for private methods.
- Added JSDoc comments in **10 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts [hunk 1]
```diff
@@ -81,6 +91,12 @@ export class ChoiceInput extends InputDialog {
         return super.onInitializeOptions(dc, options);
     }
 
+    /**
+     * @protected
+     * Called when input has been received, recognices choice.
```
Comment (santgr11) [inline libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts:96]:
```suggestion
     * Called when the input has been received, recognizes choice.
```

**Review by santgr11 — CHANGES_REQUESTED** [review]

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Comment by sw-joelmut** [comment]
@santgr11 @ceciliaavila, feedback applied, thanks.

**Review by iscaltritti-sw — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Review by santgr11 — APPROVED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts [hunk 2]
```diff
@@ -211,8 +211,8 @@ export abstract class InputDialog extends Dialog {
     /**
      * @protected
      * Called before an event is bubbled to its parent.
-     * @param dc The `DialogContext` for the current turn of conversation.
-     * @param event The event being raised.
+     * @param dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param event [DialogReason](xref:botbuilder-dialogs.DialogReason), the event being raised.
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts:231]:
```suggestion
     * @param event [DialogEvent](xref:botbuilder-dialogs.DialogEvent), the event being raised.
```

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts [hunk 3]
```diff
@@ -111,9 +111,9 @@ export abstract class InputDialog extends Dialog {
 
     /**
      * Called when the dialog is started and pushed onto the dialog stack.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/input/inputDialog.ts:126]: Should we use the Dialog class reference here?, more lines with this point on this file.

File: libraries/botbuilder-dialogs-adaptive/src/input/oauthInput.ts [hunk 4]
```diff
@@ -78,9 +78,9 @@ export class OAuthInput extends InputDialog {
 
     /**
      * Called when a prompt dialog is pushed onto the dialog stack and is being activated.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/input/oauthInput.ts:108]: Should we use the Dialog class reference here?, more lines with this point on this file.

**Review by sw-joelmut — COMMENTED** [review]
Changes

**Comment by denscollo** [comment]
@sw-joelmut @ceciliaavila what is the status of this PR? I see that there're changes request, and comments left

**Comment by sw-joelmut** [comment]
> 
> 
> @sw-joelmut @ceciliaavila what is the status of this PR? I see that there're changes request, and comments left

I'll take a look

**Comment by sw-joelmut** [comment]
@ceciliaavila @denscollo @iscaltritti-sw feedback applied, thanks.

**Review by ceciliaavila — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**